### PR TITLE
[554]Added sample perf-test configs for faiss-ivf, faiss-ivfpq, lucene-hnsw

### LINKED ISSUE #554: Add the perf test configs for different K-NN algo
Add the perf test configs for different K-NN algo in k-NN repo, so that we don't need to maintain them offline and it can updated and can be used for future perf testing.

### PR DESCRIPTION
**Pull request by navneet1v**
### Description
Added sample perf-test configs for faiss-ivf, faiss-ivfpq, lucene-hnsw. As offline configs get stale very easily this is a good way to maintain the perf test configs.

Signed-off-by: Navneet Verma <navneev@amazon.com>

 
### Issues Resolved
#554
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: benchmarks/perf-tool/sample-configs/lucene-hnsw/index.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "settings": {
+    "index": {
+      "knn": true,
+      "number_of_shards": 24,
+      "number_of_replicas": 1,
```
Comment (martin-gaievski) [inline benchmarks/perf-tool/sample-configs/lucene-hnsw/index.json:6]: I see this is different from config for nmslib,  num of replicas and `ef_construction` are different https://github.com/opensearch-project/k-NN/blob/main/benchmarks/perf-tool/sample-configs/nmslib-sift-hnsw/index-spec.json. Do we need to keep all configs identical or it's done this way on purpose?

**Review by jmazanec15 — COMMENTED** [review]
Instead of putting in "sample-configs" folder, could we create a new folder called "release-configs"?

**Comment by navneet1v** [comment]
> Instead of putting in "sample-configs" folder, could we create a new folder called "release-configs"?

done

File: benchmarks/perf-tool/sample-configs/lucene-hnsw/index.json (hunk 1 above)
Comment (navneet1v) [reply benchmarks/perf-tool/sample-configs/lucene-hnsw/index.json]: These configs are picked from the S3 bucket, which we were using to do the perf tests. I didn't make any changes.

File: benchmarks/perf-tool/release-configs/lucene-hnsw/index.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "settings": {
+    "index": {
+      "knn": true,
+      "number_of_shards": 24,
+      "number_of_replicas": 1,
+      "knn.algo_param.ef_search": 256
```
Comment (martin-gaievski) [inline benchmarks/perf-tool/release-configs/lucene-hnsw/index.json:7]: nit: this line is redundant, ef_search is ignored for lucene and value of _k_ is used instead

Comment (navneet1v) [reply benchmarks/perf-tool/release-configs/lucene-hnsw/index.json]: done

**Review by martin-gaievski — APPROVED** [review]

**Review by jmazanec15 — APPROVED** [review]

**Pull request merged**